airbyte-webapp/src/components/JobItem/components/Logs.tsx

```diff
@@ -57,4 +61,41 @@ const Logs: React.FC<LogsProps> = ({ logsArray }) => {
   );
 };
 
+/**
+ * Matching the log's line number by time makes the following assumptions:
+ * 1. The log's lines are already ordered by time
+ * 2. The timestamps used are in the same timezone
+ * 3. The error is closer to the end of the log file than the beginning
+ * 4. Lines are matched by the start of a line with "YYYY-MM-DD HH:mm:ss" format, like "2022-05-17 23:00:19 DEBUG I am a log message"
+ */
+const getMatchingLineNumbers = (matchTimestamp: number | undefined, lines: string[] | undefined) => {
+  if (!matchTimestamp || !lines || lines.length === 0) {
+    return [];
+  }
+  const flooredMatchTimestamp = Math.floor(matchTimestamp / 1000) * 1000;
+
+  const resolutionOffset = 1000; // the resolution of the timestamps is in seconds
+  const matchingLineNumbers: number[] = [];
+  const matcher = /^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1]) (2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9]/;
+
+  let lineCounter = lines.length - 1;
+  while (lineCounter >= 0) {
+    const timeString = lines[lineCounter].match(matcher);
+    if (timeString) {
+      const datetime = Date.parse(`${timeString[0].replace(" ", "T")}Z`);
```

Comment:
We should try to avoid `Date.parse`. We already use day.js as a dependency for date handling (since `Date` in JS is pretty much the same as in Java - before jodatime made it mostly into the language - try to avoid it :D). We should leverage that for [parsing](https://day.js.org/docs/en/parse/string-format) the date here, which would also give us proper control of the format.

Reply:
Done! Was able to remove the custom regexp too!